### PR TITLE
Database design

### LINKED ISSUE #26: Database choice & Model
We need to decide on the database we are going to use
As well as the model schema for how we are going to model the data

### PR DESCRIPTION
**Pull request by TroyKomodo**
## Proposed changes

The design for the backend database.

The database of choice is postgresql and we are going to be using sqlx in rust as our driver.

closes #26 

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation Update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/ScuffleTV/scuffle/blob/main/CONTRIBUTING.md) doc
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

## Further comments

Lets review the database design here for how we can go about modeling the data

### CODE REVIEW

**Review by treuks — APPROVED** [review]

**Review by PandaDex — APPROVED** [review]

**Comment by TroyKomodo** [comment]
@treuks @PandaDex why did u guys approve this?

Its not even ready to merge i want you guys to look at the structure and see if it makes sense

**Comment by treuks** [comment]
Oh lol, the structure is alright, yeah, that's what I'm approving

**Comment by intro23** [comment]
are things like payment events not being considered for the initial db/system design? Might want consider it for things like ban user interactions with payment objects or if you want the channel owner would like to handle that if on a case by case basis.

**Comment by TroyKomodo** [comment]
@intro23 payments are currently not considered at all. However stuff like that will come later

**Comment by TroyKomodo** [comment]
@ScuffleTV/reviewers when you test this PR, please try migrate a database and see if all the types in common match that which are in the DB

**Review by treuks — APPROVED** [review]
All of the DB types corresponded with the comments and rust types.
Seems good to me.

**Comment by treuks** [comment]
Also, for the testers, the DB doesn't persist because it's a debug DB. So, the migrations running on `db-up` is intended.

File: backend/api/Cargo.toml [hunk 1]
```diff
@@ -14,3 +14,4 @@ tokio = { version = "1.25.0", features = ["full"] }
 serde = { version = "1.0.152", features = ["derive"] }
 hyper = { version = "0.14.24", features = ["full"] }
 common = { path = "../../common" }
+sqlx = { version = "0.6.2", features = ["postgres", "runtime-tokio-rustls", "json", "chrono"] }
```
Comment (IS2511) [inline backend/api/Cargo.toml:17]: At least one newline at the end of file is convention.

File: common/Cargo.toml [hunk 2]
```diff
@@ -11,3 +11,7 @@ tracing = "0.1.37"
 anyhow = "1.0.69"
 serde = { version = "1.0.152", features = ["derive"] }
 tracing-subscriber = { version = "0.3.16", features = ["fmt", "env-filter", "json"] }
+chrono = "0.4.23"
+bitmask-enum = "2.1.0"
+async-trait = "0.1.64"
+sqlx = { version = "0.6.2", features = ["postgres"] }
```
Comment (IS2511) [inline common/Cargo.toml:17]: At least one newline at the end of file is convention.

**Review by 4w3Official — APPROVED** [review]
LGTM

**Review by SongoMen — APPROVED** [review]
Everything works

**Review by IS2511 — APPROVED** [review]

**Pull request merged**